zappa/core.py

```diff
@@ -2930,6 +2936,15 @@ def get_scheduled_event_name(self, event, function, lambda_name, index=0):
             # prefix all entries bar the first with the index
             # Related: https://github.com/Miserlou/Zappa/pull/1051
             name = "{}-{}".format(index, name)
```

Comment:
What does this `index` and `name` resolve to?

I thought it was function_name and stage, but the variable names here seem to suggest something else.

Reply:
My understanding is that the process is to avoid duplicate names if there are multiple expressions.
https://github.com/zappa/Zappa/issues/1036

example:
```
{
...
"events": [
            {
                "function": "application.utils.cron.some_func",
                "expressions": [
                    "cron(*/5 22-23 ? * SUN-THU *)",
                    "cron(*/5 0-8 ? * MON-FRI *)"
                ]
            }
        ]
...
}
```

Reply:
This may be an existing issue, but does the _index_ take intoaccount the `STAGE`?  

It appears for this code that the resulting rule name will be the same across all defined stages.

Reply:
> It appears from this code that the resulting rule name will be the same across all defined stages.

I understand it the same way.